src/snapshot/resource_checksum_hash.rs

```diff
@@ -30,55 +30,50 @@ use crate::{ChecksumFlag, ChecksumPart, Rollback, SaveWorld, SaveWorldSet};
 /// app.add_plugins(ResourceChecksumHashPlugin::<BossHealth>::default());
 /// # }
 /// ```
-pub struct ResourceChecksumHashPlugin<R>
-where
-    R: Resource + Hash,
-{
-    _phantom: PhantomData<R>,
+pub struct ResourceChecksumHashPlugin<R: Resource>(pub for<'a> fn(&'a R) -> u64);
+
+fn default_hasher<R: Resource + Hash>(resource: &R) -> u64 {
+    let mut hasher = bevy::utils::FixedState.build_hasher();
+    resource.hash(&mut hasher);
+    hasher.finish()
 }
 
 impl<R> Default for ResourceChecksumHashPlugin<R>
 where
     R: Resource + Hash,
 {
     fn default() -> Self {
-        Self {
-            _phantom: default(),
-        }
+        Self(default_hasher::<R>)
     }
 }
 
-impl<R> ResourceChecksumHashPlugin<R>
+impl<R> Plugin for ResourceChecksumHashPlugin<R>
```

Comment:
I don't think this struct have anything specific to "hashing" in it anymore, it just runs a function that returns a checksum?

Maybe rename?

`ResourceChecksumPlugin`?

Reply:
Agreed. I've also updated the module/file names to match that.